packages/teams-js/src/private/copilot.ts

```diff
@@ -41,12 +48,49 @@ export namespace copilot {
      *
      * @throws Error if {@linkcode app.initialize} has not successfully completed
      */
-    export function getEligibilityInfo(): AppEligibilityInformation {
+    export async function getEligibilityInfo(): Promise<AppEligibilityInformation> {
       ensureInitialized(runtime);
       if (!isSupported()) {
-        throw errorNotSupportedOnPlatform;
+        throw new Error(`Error code: ${errorNotSupportedOnPlatform.errorCode}, message: Not supported on platform`);
       }
-      return runtime.hostVersionsInfo!.appEligibilityInformation!;
+
+      // Return the eligibility information if it is already available
+      if (runtime.hostVersionsInfo?.appEligibilityInformation) {
+        copilotLogger('Eligibility information is already available on runtime.');
+        return runtime.hostVersionsInfo!.appEligibilityInformation;
+      }
+
+      copilotLogger('Eligibility information is not available on runtime. Requesting from host.');
+      // Send message to host SDK to get eligibility information
+      const response = await sendAndUnwrap<AppEligibilityInformation | SdkError>(
+        getApiVersionTag(copilotTelemetryVersionNumber, ApiName.Copilot_Eligibility_GetEligibilityInfo),
+        'copilot.eligibility.getEligibilityInfo',
+      );
+
+      if (isSdkError(response)) {
+        throw new Error(
+          `Error code: ${response.errorCode}, message: ${response.message ?? 'Failed to get eligibility information from the host.'}`,
+        );
+      }
+      // validate response
+      if (!isEligibilityInfoValid(response)) {
+        throw new Error('Error deserializing eligibility information');
+      }
+      return response;
+    }
+
+    function isEligibilityInfoValid(eligibilityInfo: AppEligibilityInformation): boolean {
```

Comment:
Even though it was always expected from hubs to not send undefined for eligibilityInfo properties, but we were not enforcing it in code in case they send undefined. Do you think we should inform them about this change?

Reply:
Sure, I can do that. It's also relatively easy to adjust this validation (vs rolling out host side changes) if we need to change it in the future.